song-server/src/main/java/org/icgc/dcc/song/server/exceptions/ServerExceptionHandler.java

```diff
@@ -37,6 +39,22 @@
     return songError.getResponseEntity();
   }
 
+  @ExceptionHandler(AccessDeniedException.class)
+  public ResponseEntity<String> handleAccessDeniedException(HttpServletRequest request, HttpServletResponse response,
+      AccessDeniedException ex){
+    val error = new SongError();
+    error.setRequestUrl(generateRequestUrlWithParams(request));
```

Comment:
would it be possible to use a builder pattern for a SongError.
With lombok it should be as easy as `@builder`